geotiepoints/tests/test_modisinterpolator.py

```diff
@@ -29,86 +34,136 @@
-
-        satz5 = satz1[2::5, 2::5]
-        lons, lats = modis_5km_to_1km(lon5, lat5, satz5)
-        lons = lons + 180
-        lons = xr.where(lons > 180, lons - 360, lons)
-        self.assertTrue(np.allclose(orig_lon, lons, atol=1e-2))
-        self.assertTrue(np.allclose(lat1, lats, atol=1e-2))
+def _to_dask(arr):
+    return da.from_array(arr, chunks=4096)
+
+
+def _to_da(arr):
+    return xr.DataArray(_to_dask(arr), dims=['y', 'x'])
+
+
+def _load_h5_geo_vars(*var_names):
+    h5f = h5py.File(FILENAME_DATA, 'r')
+    return tuple(h5f[var_name] for var_name in var_names)
+
+
+def load_1km_lonlat_as_numpy():
+    lon1, lat1 = _load_h5_geo_vars('lon_1km', 'lat_1km')
+    return lon1[:], lat1[:]
+
+
+def load_1km_lonlat_as_dask():
+    lon1, lat1 = _load_h5_geo_vars('lon_1km', 'lat_1km')
+    return _to_dask(lon1), _to_dask(lat1)
+
+
+def load_1km_lonlat_as_xarray_dask():
+    lon1, lat1 = _load_h5_geo_vars('lon_1km', 'lat_1km')
+    return _to_da(lon1), _to_da(lat1)
+
+
+def load_1km_lonlat_satz_as_xarray_dask():
+    lon1, lat1, satz1 = _load_h5_geo_vars('lon_1km', 'lat_1km', 'satz_1km')
+    return _to_da(lon1), _to_da(lat1), _to_da(satz1)
+
+
+def load_5km_lonlat_satz1_as_xarray_dask():
+    lon1, lat1, satz1 = _load_h5_geo_vars('lon_1km', 'lat_1km', 'satz_1km')
+    lon5 = lon1[2::5, 2::5]
+    lat5 = lat1[2::5, 2::5]
+    satz5 = satz1[2::5, 2::5]
+    return _to_da(lon5), _to_da(lat5), _to_da(satz5)
+
+
+def load_l2_5km_lonlat_satz1_as_xarray_dask():
+    lon1, lat1, satz1 = _load_h5_geo_vars('lon_1km', 'lat_1km', 'satz_1km')
+    lon5 = lon1[2::5, 2:-5:5]
+    lat5 = lat1[2::5, 2:-5:5]
+    satz5 = satz1[2::5, 2:-5:5]
+    return _to_da(lon5), _to_da(lat5), _to_da(satz5)
+
+
+def load_500m_lonlat_expected_as_xarray_dask():
+    h5f = h5py.File(FILENAME_DATA, 'r')
+    lon500 = _to_da(h5f['lon_500m'])
+    lat500 = _to_da(h5f['lat_500m'])
+    return lon500, lat500
+
+
+def load_250m_lonlat_expected_as_xarray_dask():
+    h5f = h5py.File(FILENAME_DATA, 'r')
+    lon250 = _to_da(h5f['lon_250m'])
+    lat250 = _to_da(h5f['lat_250m'])
+    return lon250, lat250
+
+
+def assert_geodetic_distance(
+        lons_actual: np.ndarray,
+        lats_actual: np.ndarray,
+        lons_desired: np.ndarray,
+        lats_desired: np.ndarray,
+        max_distance_diff: float,
+) -> None:
+    """Check that the geodetic distance between two sets of coordinates is smaller than a threshold.
+
+    Args:
+        lons_actual: Longitude array produced by interpolation being tested.
+        lats_actual: Latitude array produced by interpolation being tested.
+        lons_desired: Longitude array of expected/truth coordinates.
+        lats_desired: Latitude array of expected/truth coordinates.
+        max_distance_diff: Limit of allowed distance difference in meters.
+
+    """
+    g = Geod(ellps="WGS84")
+    _, _, dist = g.inv(lons_actual, lats_actual, lons_desired, lats_desired)
+    np.testing.assert_array_less(dist, max_distance_diff)  # meters
```

Comment:
For clarity, I think we should provide a custom error message to `assert_array_less`.

Reply:
```
        err_msg=f"Coordinates are greater than {max_distance_diff} geodetic "
                "meters from the expected coordinates.")
```

Done.